crates/sui/src/client_commands.rs

```diff
@@ -2167,3 +2213,45 @@ impl Display for SwitchResponse {
         write!(f, "{}", writer)
     }
 }
+
+/// Request tokens from the Faucet for the given address
+pub async fn request_tokens_from_faucet(
+    address: SuiAddress,
+    url: String,
+) -> Result<(), anyhow::Error> {
+    let address_str = address.to_string();
+    let json_body = json![{
+        "FixedAmountRequest": {
+            "recipient": &address_str
+        }
+    }];
+
+    // make the request to the faucet JSON RPC API for coin
+    let client = reqwest::Client::new();
+    let resp = client
+        .post(&url)
+        .header("Content-Type", "application/json")
+        .json(&json_body)
+        .send()
+        .await?;
+    println!("Requested gas from faucet: {url}");
+    println!(
+        "Faucet request for address {address_str} has status: {}",
+        resp.status()
+    );
+    if resp.status() == 429 {
+        bail!("Faucet received too many requests from this ip address. Please try again after 60 minutes.");
+    }
+    println!("Waiting for the faucet to complete the gas request...");
+
+    let faucet_resp: FaucetResponse = resp.json().await?;
+
+    let task_id = if let Some(err) = faucet_resp.error {
+        bail!("Faucet request was unsuccessful. Error is {err:?}")
```

Comment:
```suggestion
        bail!("Faucet request was unsuccessful: {err}")
```

Don't use debug output for errors -- most of the time the `impl Display` for the error type displays a human-readable message, and we should use that.  In the cases it doesn't, we should fix that.

Reply:
Oops, my bad. Thanks for spotting this!